### DISCUSSION TITLE
Weather widget violating free tier

### DISCUSSION THREAD

**Discussion by Ryuno-Ki** [streetturtle/awesome-wm-widgets#363] [Q&A]
This week I received an email by OpenWeatherMap informing me about hitting the limit of 1.000 requests/day.

I'm using the weather widget as-is up until now. They shared a screenshot of the stats and it appears to be most the times within range.
However, in order to track the usage, I was told to implement a counter on my end (help welcome. I'm willing to PR if there's interest, but I don't know where to start).

Now my laptop is often either on (with me working on it) or turned off entirely. I have a few occassions, where I simply lock it - sometimes for days (no, I don't want to hibernate or suspend). When I unlock it, I can observe a high CPU load (thanks to the CPU widget).
I could imagine, that it also yields a burst to their API endpoint.

In order to have data at hand, I would like to get some starting points.

Am I the only one with this issue?

**Comment by streetturtle** [comment]
The default configuration of the refresh rate is 2 minutes, so per day (24 hours) it should call their API 720 times (24 * 60 / 2), which is within the limit of 1000 requests per day. 

I have discovered recently (thanks to several issues) that if you have two or more screens, and define the widget in the `s.mywibox:setup` block (as suggested in readme), Awesome will instantiate one widget per screen, meaning each instance will call OWM API, so, for example, with two screens there will be 1440 calls to the API. 

Could you please check if it's the case. 

BTW, this is very valuable feedback, thank you!

**Reply by Ryuno-Ki** [reply]
Indeed I had a second monitor attached and Awesome replicated on it. Never thought of it as a cause of trouble.

I configured my widget to hit the API every 5 minutes which would mean 24 * 60 / 5 = 480 times. Even doubled that should still remain within the tier. Might be worth to set as default?

I can open a PR (that also adds a line to the README of that widget to inform about additional screens).

**Reply by streetturtle** [reply]
I think a better way would be to instantiate the widget outside the setup function, as it doesn't make sense to have two instances.
But it should be tested first, as I am not sure how the popup widget would behave.

Regarding the rate limiting, I think the quick and dirty way would be to simply log a timestamp every time the API is called (line 501 for example) in some file, like 'awful.spawn(echo date +%s >> somefile.log)'.

**Reply by Ryuno-Ki** [reply]
I'll report back once I tried this for a couple days.
